src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,39 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonalArray = new int[Math.min(matrix.length, matrix[0].length)];
```

Comment:
let's make it esier)
```suggestion
        int[] mainDiagonalArray = new int[matrix.length];
```

Reply:
but in this case we are counting only amount of columns, aren't we? And if rows<columns then we create additional empty cells in array. Why do we need that?